src/Timeline.js

```diff
@@ -74,24 +75,37 @@ export default class Timeline extends Axis {
               ? event.selection 
               : [event.selection[0] + buttonDomain, event.selection[1] - buttonDomain].sort((a, b) => a - b);
 
-      const domain = (this._brushing ? selection
-        : this._buttonBehaviorCurrent === "buttons" 
-          ? [event.sourceEvent.offsetX, event.sourceEvent.offsetX] : [event.selection[0], event.selection[0]])
-        .map(this._d3Scale.invert)
-        .map(Number);
-
-      const ticks = this._availableTicks.map(Number);
-
-      domain[0] = date(closest(domain[0], ticks));
-      domain[1] = date(closest(domain[1], ticks));
+      const domain = this._buttonBehaviorCurrent === "ticks"
```

Comment:
this logic of determining the `pixelDomain` is getting rather large, and being used in 3 places. move this code out of these 3 functions into it's on method of the class called `_pixelDomain`